alphadia/workflow/optimization.py

```diff
@@ -654,7 +655,7 @@ def update_with_extraction(self, feature_df, fragment_df):
         self.feature_dfs += [feature_df]
         self.fragment_dfs += [fragment_df]
 
-        self.total_precursors = len(self.features_df)
+        self.total_precursors = self.features_df.precursor_idx.nunique()
```

Comment:
As every precursor is only in the library once:
`self.total_precursors += feature_df.precursor_idx.nunique()`

Is it correct that we track the precursor here and not the elution_groups?

Reply:
This will require resetting the total precursors value at the same time as the feature_dfs list, but I think it is a good idea so will do so. 

You are correct; it would be better to use the elution groups here.